pygenomeworks/genomeworks/cudapoa/cudapoa.pyx

```diff
@@ -132,14 +128,22 @@ cdef class CudaPoaBatch:
         cdef int32_t mx_seq_per_poa = max_sequences_per_poa
         cdef int32_t mx_consensus_sz = \
             2 * max_sequence_size if max_consensus_size is None else max_consensus_size
-        cdef int32_t mx_nodes_per_w = \
-            3 * max_sequence_size if max_nodes_per_graph is None else max_nodes_per_graph
-        cdef int32_t mx_nodes_per_w_banded = \
-            4 * max_sequence_size if max_nodes_per_graph_banded is None else max_nodes_per_graph_banded
+        cdef int32_t mx_nodes_per_w
+        cdef BandMode batch_band_mode
+        if (band_mode == "full_band"):
+            batch_band_mode = BandMode.full_band
```

Comment:
I'm a bit confused here, what happens to two other parameters `matrix_graph_dimension` and `matrix_sequence_dimension` (these two are used to define buffer space for score matrix). I see it was not defined before your change as well, that's why I'm confused.

Reply:
This results in the second constructor being called: https://github.com/r-mafi/ClaraGenomicsAnalysis/blob/abpoa_modified/cudapoa/include/claraparabricks/genomeworks/cudapoa/batch.hpp#L119

There's no handle to modify those values directly from here.

Reply:
that's a bug then. In `static_banded`, `matrix_sequence_dimension` = `band_width + 8`, and in `adaptive_band`, it is defined as double of that. The second constructor in the cpp needs to change to include `matrix_sequence_dimension` as well.

Reply:
I added `matrix_sequence_dimension` to the constructor. Should `matrix_graph_dimension` be added as well? We don't have any special treatment for that param.

Reply:
no, before this PR, `matrix_graph_dimension` and `matrix_sequence_dimension` were equal to `max_nodes_per_graph` and `max_sequence_size` in order, but after this PR, we changed `matrix_sequence_dimension` to be double size for adaptive. But `matrix_graph_dimension` does not have any special treatment.